File: 2021/python/day01.py
numbers = [int(i.strip()) for i in open("../input/day01.input", encoding="utf-8").readlines()]
avg = [numbers[i] + numbers[i-1] + numbers[i-2] for i in range(2, len(numbers))]
print("Answer A:", sum([a < b for a, b in zip(numbers, numbers[1:])]))
print("Answer B:", sum([a < b for a, b in zip(avg, avg[1:])]))

File: 2021/golang/day01.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func Day01() {
	file, _ := os.Open("../input/day01.input")
	defer file.Close()
	scanner := bufio.NewScanner(file)
	var digits []int

	for scanner.Scan() {
		digit, _ := strconv.Atoi(scanner.Text())
		digits = append(digits, digit)
	}

	partA := 0
	for i := 1; i < len(digits); i++ {
		if digits[i] > digits[i-1] {
			partA++
		}
	}

	partB := 0
	for i := 3; i < len(digits); i++ {
		if digits[i] > digits[i-3] {
			partB++
		}
	}

	fmt.Println("Answer A:", partA)
	fmt.Println("Answer B:", partB)
}
